java/src/by/gsu/epamlab/task3/Purchase.java

```diff
@@ -0,0 +1,57 @@
+package by.gsu.epamlab.task3;
+
+public class Purchase implements Comparable<Purchase> {
+    public static final String NAME = "Bicycle";
+    public static final int PRICE = 100000;
+    private int quantity;
+    private int discount;
+    private WeekDay weekDay;
+
+    public Purchase(int quantity, int discount, WeekDay weekDay) {
+        this.quantity = quantity;
+        this.discount = discount;
+        this.weekDay = weekDay;
+    }
+
+    public Purchase() {
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public int getDiscount() {
+        return discount;
+    }
+
+    public void setDiscount(int discount) {
+        this.discount = discount;
+    }
+
+    public WeekDay getWeekDay() {
+        return weekDay;
+    }
+
+    public void setWeekDay(WeekDay weekDay) {
+        this.weekDay = weekDay;
+    }
+
+    public int getCost() {
```

Comment:
This shouldn't be "int" because discount should be double.
And operation is incorrect because you always round to small number (in your code 10.95 is 10, not 11 )